lib/FreeRTOS-Plus-TCP/source/FreeRTOS_DNS.c

```diff
@@ -446,7 +446,8 @@ TickType_t xIdentifier = 0;
 	/* Generate a unique identifier. */
 	if( 0 == ulIPAddress )
 	{
-		xIdentifier = ( TickType_t )ipconfigRAND32( );
+		/* DNS identifiers are 16-bit. */
+		xIdentifier = ( TickType_t )( ipconfigRAND32( ) & 0xffff );
```

Comment:
Yes, but in the context of this function, the identifier is a TickType_t, so the mask seems risky. An alternative would be to change the type of xIdentifier, as well as the prototype of the routine below that it gets passed to.